flashcards/web/src/modules/AddDeck/PreviewImageLibrary.tsx

```diff
@@ -0,0 +1,93 @@
+import { CircularProgress, Typography } from "@mui/material";
+import { useEffect, useState } from "react";
+import { faker } from "@faker-js/faker";
+import { Image } from "./ImagePicker";
+
+interface LibraryImageListProps {
+    setShowModal: (showModal: boolean) => void;
+    images: Image["images"];
+    setIsLoading: (isLoading: boolean) => void;
+    isLoading: boolean;
+}
+
+function PreviewImageLibrary({ images, isLoading, setIsLoading, setShowModal }: LibraryImageListProps) {
+    const [previewImages, setPreviewImages] = useState<string[]>([]);
+
+    useEffect(() => {
```

Comment:
Following ReactQuery logic, I'd extract this into separate hook (even if it's supposed to be mocked for now).